Source/WebCore/animation/ElementAnimationRareData.cpp

```diff
@@ -54,4 +54,16 @@ void ElementAnimationRareData::setAnimationsCreatedByMarkup(CSSAnimationCollecti
     m_animationsCreatedByMarkup = WTFMove(animations);
 }
 
+void ElementAnimationRareData::setLastStyleChangeEventStyle(std::unique_ptr<const RenderStyle>&& style)
+{
+    if (m_keyframeEffectStack) {
+        auto previousStyleChangeEventStyle = std::exchange(m_lastStyleChangeEventStyle, WTFMove(style));
+        if (previousStyleChangeEventStyle != m_lastStyleChangeEventStyle)
```

Comment:
Don't you want to test this before exchanging, in case you can avoid the exchange?

Reply:
Addressed in e0da69b142fb.